modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/reducers/fieldSetReducer.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {
+	FieldSetUtil,
+	FieldSupport,
+	SettingsContext,
+} from 'dynamic-data-mapping-form-builder';
+import {EVENT_TYPES, PagesVisitor} from 'dynamic-data-mapping-form-renderer';
+
+import {SIDEBAR} from '../eventTypes';
+
+export default (state, action, config) => {
+	switch (action.type) {
+		case SIDEBAR.FIELDSET.UPDATE: {
+			const {dataDefinitionId, editingDataDefinitionId} = state;
+			const fieldSets = action.payload.fieldSets.filter(
+				(fieldSet) =>
+					fieldSet.id !== dataDefinitionId &&
+					fieldSet.id !== editingDataDefinitionId
+			);
+
+			return {
+				...state,
+				fieldSets,
+			};
+		}
+		case EVENT_TYPES.FIELD_SET.ADD: {
```

Comment:
As you moved this event here so it makes no sense to keep the event in the `core` anymore, we can delete it and add this event type on the side of the taglib only.